From c9b2b18e7225a757fc10c20531c12c3bca9fd484 Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Tue, 30 Jul 2013 13:26:06 +0800
Subject: Use SW memory on the output of Media SDK based SW OpenMAX decoder plug-ins

BZ: 120796

Use SW memory on the output of Media SDK based SW OpenMAX decoder plug-ins

Category: feature differentiation
Domain: Video.Media-Video Codec
Origin: internal
Upstream-Candidate: no

Change-Id: I6f38d4b9ea928f034b0d6e72dd72b2006d02aa58
Orig-Change-Id: Ia8ed378420e9454a3b4c107eabea8b7dddbf2b59
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    3 ++-
 media/libstagefright/OMXCodec.cpp      |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index cd2b7b1..8f1f68e 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1219,7 +1219,8 @@ void AwesomePlayer::initRenderer_l() {
     setVideoScalingMode_l(mVideoScalingMode);
     if (USE_SURFACE_ALLOC
             && !strncmp(component, "OMX.", 4)
-            && strncmp(component, "OMX.google.", 11)) {
+            && strncmp(component, "OMX.google.", 11)
+            && strncmp(component, "OMX.Intel.sw_vd", 15)) {
         // Hardware decoders avoid the CPU color conversion by decoding
         // directly to ANativeBuffers, so we must use a renderer that
         // just pushes those buffers to the ANativeWindow.
diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index cb15522..1032c7b 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -1538,7 +1538,8 @@ OMXCodec::OMXCodec(
       mLeftOverBuffer(NULL),
       mPaused(false),
       mNativeWindow(
-              (!strncmp(componentName, "OMX.google.", 11))
+              (!strncmp(componentName, "OMX.google.", 11)
+              || !strncmp(componentName, "OMX.Intel.sw_vd", 15))
                         ? NULL : nativeWindow) {
     mPortStatus[kPortIndexInput] = ENABLED;
     mPortStatus[kPortIndexOutput] = ENABLED;
-- 
1.7.4.1

